### PR TITLE
Channel sends typed responses

### PR DESCRIPTION
**Pull request by t-bast**
Instead of sending strings, channel now sends typed responses.
This is more future-proof when we want to add data to those responses.

type-all-the-things

TODO:

- [x] E2E tests to verify nothing funky is happening.

### CODE REVIEW

**Comment by t-bast** [comment]
Looks like I have to do some json serialization magic to make it smooth in the CLI, will add that in a separate commit.

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala [hunk 1]
```diff
@@ -144,19 +145,19 @@ class EclairImpl(appKit: Kit) extends Eclair {
       pushMsat = pushAmount_opt.getOrElse(0 msat),
       fundingTxFeeratePerKw_opt = fundingFeerateSatByte_opt.map(feerateByte2Kw),
       channelFlags = flags_opt.map(_.toByte),
-      timeout_opt = Some(openTimeout))).mapTo[String]
+      timeout_opt = Some(openTimeout))).mapTo[ChannelCommandResponse].map(_.toString)
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: You could return a `ChannelCommandResponse` and define a json deserializer that does a `toString`. Not sure if that'd be better.

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala [hunk 2]
```diff
@@ -84,13 +83,20 @@ object Channel {
...
 
   // @formatter:off
-  sealed trait ChannelError
-  case class LocalError(t: Throwable) extends ChannelError
-  case class RemoteError(e: Error) extends ChannelError
+  sealed trait ChannelOpenError
+  case class LocalError(t: Throwable) extends ChannelOpenError
+  case class RemoteError(e: Error) extends ChannelOpenError
+
+  sealed trait ChannelCommandResponse
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: Commands are defined in `ChannelTypes.scala` and responses in `Channel.scala`.

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala [hunk 3]
```diff
@@ -1733,7 +1739,7 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
   /**
    * This function is used to return feedback to user at channel opening
    */
-  def replyToUser(message: Either[Channel.ChannelError, String]): Unit = {
+  def replyToUser(message: Either[ChannelOpenError, ChannelCommandResponse]): Unit = {
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: Maybe this method should be renamed to make it clear this is used at the channel opening.

File: eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala (hunk 1 above)
Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: It probably makes sense to return a `ChannelCommandResponse` and push the conversion to string (which is lossy) closer to the boundary of the program.

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: Yes. The disadvantage being that there is a symmetry problem given that this function doesn't take a `Command`. I believe we are in bikeshed territory.

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala]: True, but I think it's quite ok there :)

**Comment by pm47** [comment]
Needs merge/rebase

**Pull request merged**